src/astx/flows.py

```diff
@@ -418,3 +418,104 @@ def get_struct(self, simplified: bool = False) -> ReprStruct:
         }
 
         return self._prepare_struct(key, value, simplified)
+
+
+@public
+@typechecked
+class CaseStmt(StatementType):
+    """AST class for a branch in a switch/match expression."""
+
+    condition: Optional[Expr] = None
+    body: Expr
+    default: bool = False
+
+    def __init__(
+        self,
+        body: Expr,
+        condition: Optional[Expr] = None,
+        default: bool = False,
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+        parent: Optional[ASTNodes] = None,
+    ) -> None:
+        """Initialize the CaseStmt instance."""
+        super().__init__(loc=loc, parent=parent)
+        self.condition = condition
+        self.body = body
+        self.default = default
+        self.kind = ASTKind.CaseStmtKind
+
+        if self.default is False and self.condition is None:
+            raise ValueError(
+                "Condition must be provided for non-default branches."
+            )
+
+        if self.default is True and self.condition is not None:
+            raise ValueError(
+                "Condition must NOT be provided for default branches."
+            )
+
+    def __str__(self) -> str:
+        """Return a string representation of the object."""
+        return f"CaseStmt[{self.condition} => {self.body}]"
+
+    def get_struct(self, simplified: bool = False) -> ReprStruct:
+        """Return the AST structure of the object."""
+        key = "CASE-STMT"
+        condition_dict = (
+            {"condition": {"default": []}}
+            if self.condition is None
+            else {"condition": self.condition.get_struct(simplified)}
+        )
+        value = {
+            **cast(DictDataTypesStruct, condition_dict),
+            "body": self.body.get_struct(simplified),
+        }
+        return self._prepare_struct(key, value, simplified)
+
+
+@public
+@typechecked
+class SwitchStmt(StatementType):
+    """AST class for switch/match expressions based on Rust's match syntax."""
+
+    value: Expr
+    cases: list[CaseStmt] | ASTNodes[CaseStmt]
+
+    def __init__(
+        self,
+        value: Expr,
+        cases: list[CaseStmt] | ASTNodes[CaseStmt],
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+        parent: Optional[ASTNodes] = None,
+    ) -> None:
+        """Initialize the SwitchStmt instance."""
+        super().__init__(loc=loc, parent=parent)
+        self.value = value
+        self.cases = cases
```

Comment:
in the case it is a list of casestmt, we need to convert it to astnodes of casestmt